src/util/mod.rs

```diff
@@ -33,10 +35,32 @@ pub fn timestamp_to_naive(ts: i64, ms: u32) -> NaiveDateTime {
 
 pub fn make_client() -> Client<HttpsConnector<HttpConnector>> {
     let https = HttpsConnector::new();
+    // let mut http = HttpConnector::new();
+    // http.set_connect_timeout(Some(std::time::Duration::from_secs(5)));
+    // let https = HttpsConnector::new_with_connector(http);
 
     Client::builder().build::<_, hyper::Body>(https)
 }
 
+pub async fn request_with_timeout(
```

Comment:
Shall we put timeout as one of the input params for different upstreams?

Reply:
Agree, but good for now to observe if issue is mitigated or not.